dht/table.go

```diff
@@ -26,6 +26,9 @@ type Table interface {
 	// PeerAddress returns the network address associated with the given peer.
 	PeerAddress(id.Signatory) (wire.Address, bool)
 
+	RegisterIP(id.Signatory, string)
+	IP(id.Signatory) (string, bool)
```

Comment:
We also need a method for deleting IP addresses, otherwise it will grow unbounded.

Also, for consistency, we should probably call this `AddIP` as oppose to `RegisterIP`. The functionality is no different to that of `AddPeer` (but for IPs), so it is odd that it has a different naming convention.